libs/wingsdk/src/platform/platform-manager.ts

```diff
@@ -21,6 +21,42 @@ export class PlatformManager {
 
   constructor(options: PlatformManagerOptions) {
     this.platformPaths = options.platformPaths ?? [];
+    this.retrieveImplicitPlatforms();
+  }
+
+  /**
+   * Retrieve all implicit platform declarations.
+   *
+   * These are platforms that are not explicitly declared in the cli options
+   * but are implicitly available to the app.
+   *
+   * We look for platforms in the following locations:
+   * - The source directory
+   * - Any imported namespaces (provided by the wingc compiler output)
+   *
+   * To determine if a directory contains a platform, we check if it contains a file named "platform.js"
```

Comment:
I am wondering if we should use a more distinct name such as `wing-platform.js` (or `w-platform.js`, `wing_platform.js`, `platform.wing.js`, `wp.js`, `_wing_platform.js`).

@MarkMcCulloh what do you think?

Reply:
I think it'd be useful follow whatever convention we would use if these were wing files.
Other than that I don't feel super strong about a certain direction, but if we follow the theme of `main` and `test` then we'd have `platform`:

```
platform.w
whatever.platform.w

platform.js
whatever.platform.js
```

Reply:
Ill update the convention to 
```
platform.js
*.platform.js
```

Reply:
Since this is implicitly loading these files I wanted to avoid people accidentally creating a `platform.js` or `platform.w` file and struggling to understand what the F...

Also, it's important that when people see this file they immediately understand its meaning.

Something I like about @MarkMcCulloh's suggestion is that it will be possible to put multiple platform files per directory, and that's required because a platform extension only applies to a single platform model.

So maybe just use `wplatform.js` as the extension:

```
tfaws.wplatform.js
awscdk.wplatform.js
wplatform.js
redis.awscdk.wplatform.js
```

A bit of a mouthful but better than hijacking "platform" :-)

Reply:
> A bit of a mouthful but better than hijacking "platform" :-)

😁 I wish I had a better suggestion that didnt go against the grain of the other existing conventions. Ill update to go with `wplatform` and learn to love it.

Reply:
What if we always require at least two dots? `*.platform.js`. I believe this is how `test` files work as well.

I feel like it'd be hard to hit accidentally. Honestly I suspect platform files are going to be extremely common so even with `platform.js` this won't be an issue for users to understand.

Since we're likely going to be experimenting with more verbose CLI output over time, it'd be reasonable to also inform the users of which platforms are loaded which would further ease any confusion.

Sorry if it feels like I'm belaboring this point, I'm okay with `wplatform` to avoid more naming bikeshedding, but figured I'd finish the braindump.

Reply:
I think it makes sense to also support `wplatform.js` (without a prefix).

But @MarkMcCulloh if you feel strongly, I am also okay with just supporting `*.platform.js`.

@hasanaburayyan your call...

Reply:
lets roll with `wplatform` and `*.wplatform` for now, it grew on me overnight.